awx/ui_next/src/screens/Template/WorkflowJobTemplateVisualizer/Modals/NodeModals/NodeViewModal.jsx

```diff
@@ -11,56 +11,116 @@ import ContentError from '@components/ContentError';
 import ContentLoading from '@components/ContentLoading';
 import PromptDetail from '@components/PromptDetail';
 import useRequest from '@util/useRequest';
-import { JobTemplatesAPI, WorkflowJobTemplatesAPI } from '@api';
+import {
+  InventorySourcesAPI,
+  JobTemplatesAPI,
+  ProjectsAPI,
+  WorkflowJobTemplatesAPI,
+} from '@api';
+
+function getNodeType(node) {
+  const ujtType = node.type || node.unified_job_type;
+
+  let nodeType;
+  let nodeAPI;
+  switch (ujtType) {
+    case 'job_template':
+    case 'job':
+      nodeType = 'job_template';
+      nodeAPI = JobTemplatesAPI;
+      break;
```

Comment:
You could probably just return from each of these case blocks. e.g.:

```
case 'job_template':
case 'job':
  return ['job_template', JobTemplatesAPI]`
```

Reply:
Yeah, I like how we could remove the extra variables by returning in each case block.